### PR TITLE
Command line switch to override default scripts location

### PR DESCRIPTION
**Pull request by davidkelly**
Using --scripts <some directory>, you can now set the default scripts location for this session.  When doing so, we _only_ load the defaultScripts.js, nothing else you may have saved in your settings.  When then running again without the switch, we will load the scripts in your settings again.

### CODE REVIEW

**Comment by davidkelly** [comment]
Test Plan

Start interface without any switches. 
Open the running scripts dialog, and add one
Stop interface
Make a copy of the scripts directory
Start interface, but with --scripts <full path to copy of scripts directory>
Open running scripts dialog
You should see just defaultScripts.js and controllerScripts.js
Hit 'Reveal Scripts Folder' and verify it is the one you passed in on the command line
Stop interface
Start interface, with no switches on the command line
Open running scripts dialog
You should see the script you added again
Hit 'Reveal Scripts Folder' and verify it is the default again
stop interface
:beers:

**Comment by CainFoool** [comment]
Testing.

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1028/HighFidelity-Beta-PR10790-06d1ac8bf7883baefc32141f483d61f6d46e04cd.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1028/HighFidelity-Beta-PR10790-06d1ac8bf7883baefc32141f483d61f6d46e04cd.dmg

**Comment by CainFoool** [comment]
# Test Passed

Test | Result
-------- | --------
Start interface without any switches. | Pass
Open the running scripts dialog, and add one | Pass
Stop interface | Pass
Make a copy of the scripts directory | Pass
Start interface, but with --scripts <copied directory> | Pass
Open running scripts dialog | Pass
You should see just defaultScripts.js and controllerScripts.js | Pass
Hit 'Reveal Scripts Folder' and verify it is the one you passed in on the command line | Pass
Stop interface | Pass
Start interface, with no switches on the command line | Pass
Open running scripts dialog | Pass
You should see the script you added again | Pass
Hit 'Reveal Scripts Folder' and verify it is the default again | Pass
stop interface | Pass

There was a small mistake in your test plan but I fixed it, you forgot to specify the new script directory in the command line. Everything works perfect!

Let me know if you need anything else tested.

###### Intel i5 7600K @ 4Ghz, 16GB DDR4 2133Mhz Corsair Vengeance, Asus Strix GTX 1050Ti 3GB, HTC Vive

File: libraries/shared/src/PathUtils.cpp [hunk 1]
```diff
@@ -20,6 +20,8 @@
 #include <QtCore/QStandardPaths>
 #include <mutex> // std::once
 
+//QUrl PathUtils::_overriddenDefaultScriptsLocation = QUrl();
```
Comment (howard-stearns) [inline libraries/shared/src/PathUtils.cpp]: remove?

Comment (davidkelly) [inline libraries/shared/src/PathUtils.cpp]: ohh yes - thx nice catch.  Hate leaving turds like that around!

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1053/HighFidelity-Beta-PR10790-7ef50042bafc79268f5fdb729032bb14f8636e5e.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1053/HighFidelity-Beta-PR10790-7ef50042bafc79268f5fdb729032bb14f8636e5e.exe

**Pull request merged**